django/db/models/options.py

```diff
@@ -291,6 +291,8 @@ def add_field(self, field, virtual=False):
         # self.many_to_many.
         if virtual:
             self.virtual_fields.append(field)
+            if not hasattr(field, 'originating_model'):
```

Comment:
Let's avoid `hasattr` as much as possible by having a default value of `originating_model = None`.